### PR TITLE
First attempt to remove recursion from GoodsNomenclatureMapper.

### PR DESCRIPTION
**Pull request by alexdesi**
### What?
First attempt to remove recursion from GoodsNomenclatureMapper.

### Why?
To simplify the GoodsNomenclatureMapper - Still too complex anyway, that's only the first step.

### CODE REVIEW

File: app/lib/goods_nomenclature_mapper.rb [hunk 1]
```diff
@@ -52,20 +52,19 @@ def for_goods_nomenclature(ref_goods_nomenclature)
...
+  def traverse(goods_nomenclatures, current_goods_nomenclature)
+    current_good = current_goods_nomenclature
+
+    while (index_gn = goods_nomenclatures.index(current_good))
+      next_good = goods_nomenclatures[index_gn + 1]
+      return unless next_good
+
+      map_goods_nomenclatures(current_good, next_good)
+      current_good = next_good
     end
```
Comment (alexdesi) [inline app/lib/goods_nomenclature_mapper.rb:68]: this is the refactor,
the rest of the changes are made by Rubocop.

File: app/lib/goods_nomenclature_mapper.rb [hunk 2]
```diff
@@ -52,20 +52,19 @@ def for_goods_nomenclature(ref_goods_nomenclature)
...
-    unless goods_nomenclatures.index(secondary).blank?
-      next_goods_nomenclature = goods_nomenclatures[goods_nomenclatures.index(secondary) + 1]
-      if next_goods_nomenclature.present? # we are not at the end of the goods_nomenclature array
-        map_goods_nomenclatures(secondary, next_goods_nomenclature)
-        traverse(goods_nomenclatures, secondary, next_goods_nomenclature)
-      end
+  def traverse(goods_nomenclatures, current_goods_nomenclature)
+    current_good = current_goods_nomenclature
+
+    while (index_gn = goods_nomenclatures.index(current_good))
```
Comment (willfish) [inline app/lib/goods_nomenclature_mapper.rb:62]: This is basically just an `each_cons(2)` no? I've basically never seen a need for while in ruby unless the condition for exiting the loop is outside of the data-structure.

**Review by willfish — CHANGES_REQUESTED** [review]
I'm loath to make changes to this before we've added specs that cover the necessary and sufficient aspects of this class and it's api

File: app/lib/goods_nomenclature_mapper.rb (hunk 2 above)
Comment (alexdesi) [reply app/lib/goods_nomenclature_mapper.rb:62]: yeah, I thought about it, in this first step I strictly followed the logic in the recursion, but it seems to be just a each loop, I'll change it.

**Pull request closed without merging**